pywinauto/element_info.py

```diff
@@ -166,6 +166,21 @@ def filter_with_depth(elements, root, depth):
         else:
             return elements
 
+    @staticmethod
+    def get_descendants_with_depth(elem, depth=None, **kwargs):
+        """Return a list of all descendant children of the element with the specified depth"""
+        def traverse_tree(root, output_list, depth, **kwargs):
+            if depth == 0:
+                return
+            for child in root.children(**kwargs):
+                output_list.append(child)
+                next_depth = None if depth is None else depth - 1
+                traverse_tree(child, output_list, next_depth, **kwargs)
+
+        descendants = []
```

Comment:
It could be moved before the body of `walk_the_tree()` function. So passing `output_list` is not needed then. In-place function can use variables in current context. Minus one param passed on the call stack may reduce performance load.

Reply:
fixed